### DISCUSSION TITLE
C programming

### DISCUSSION THREAD

**Discussion by botk1ngOp** [ish-app/ish#1900] [Q&A]
I am new to linux …how to run c programing..?

**Comment by morgan9992** [comment]
C compiler - apk add gcc

**Comment by 0x4248** [comment]
1) Install GCC using apk add gcc
2) Get the file you want to run
3) once gcc is installed run gcc Filename.c -o out.o
4) then run ./out.o

**Reply by j33tsud0** [reply]
I have enjoyed clang. 

Eg) 

$ apk add clang 
$ apk add vim
$ vim hi.c
______________________


#include <stdio.h>
#include <stdlib.h> 

// ignore this

int main(void) { 
    if ( puts("hi, C") == EOF) {
    return EXIT_FAILURE; }
    return EXIT_SUCCESS;
}

esc,:, w, q, enter

______________________

$ cc -o hello hi.c 
$./hello 
$ hi, C
